pkg/autostart/autostart.go

```diff
@@ -0,0 +1,129 @@
+// Package manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		// Early return if error is unexpected
+		if !errors.Is(err, os.ErrExist) && !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Unit file exists, return nil
+		if errors.Is(err, os.ErrExist) {
+			return nil
+		}
+		// config/.systemd folder might be unavailable
+		if errors.Is(err, os.ErrNotExist) && hostOS == "linux" {
+			if err := os.MkdirAll(filepath.Dir(unitPath), os.ModePerm); err != nil {
+				return err
+			}
+			// directory created try to create unit file
+			if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+				return err
+			}
+		}
+	}
+	return enableDisableService("enable", hostOS, GetStartAtLoginEntryPath(hostOS, instName))
+}
+
+// DeleteStartAtLoginEntry respect host OS arch and delete unit file
+func DeleteStartAtLoginEntry(hostOS, instName string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil {
+		// Skip error log if file not present
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+	if err := enableDisableService("disable", hostOS, GetStartAtLoginEntryPath(hostOS, instName)); err != nil {
+		return err
+	}
+	return os.Remove(unitPath)
+}
+
+// GetFilePath returns the path to autostart file with respect of host
+func GetStartAtLoginEntryPath(hostOS, instName string) string {
+	var fileTmp string
+	if hostOS == "darwin" { // launchd plist
+		fileTmp = fmt.Sprintf("Library/LaunchAgents/io.lima-vm.autostart.%s.plist", instName)
```

Comment:
What should we do when an instance name contains a period?

Reply:
I think that is fine; it should just work.

Reply:
The instance name is validated here:

https://github.com/lima-vm/lima/blob/752afc08689e06d00293b3261817d5be70cb0148/pkg/store/store.go#L62

```go
const (
	maxLength  = 76
	alphanum   = `[A-Za-z0-9]+`
	separators = `[._-]`
)
```